File: shared/ontology/src/enums.py
"""Ontology 枚举定义"""
import enum


class OrderStatus(str, enum.Enum):
    pending = "pending"
    confirmed = "confirmed"
    preparing = "preparing"
    ready = "ready"
    served = "served"
    completed = "completed"
    cancelled = "cancelled"


class StoreStatus(str, enum.Enum):
    active = "active"
    inactive = "inactive"
    renovating = "renovating"
    preparing = "preparing"
    closed = "closed"


class InventoryStatus(str, enum.Enum):
    normal = "normal"
    low = "low"
    critical = "critical"
    out_of_stock = "out_of_stock"


class TransactionType(str, enum.Enum):
    purchase = "purchase"
    usage = "usage"
    waste = "waste"
    adjustment = "adjustment"
    transfer = "transfer"
    receiving = "receiving"        # 验收入库
    transfer_out = "transfer_out"  # 调拨出库
    transfer_in = "transfer_in"    # 调拨入库


class ReceivingOrderStatus(str, enum.Enum):
    draft = "draft"
    inspecting = "inspecting"
    partially_received = "partially_received"
    fully_received = "fully_received"
    rejected = "rejected"


class ReceivingItemStatus(str, enum.Enum):
    pending = "pending"
    accepted = "accepted"
    partial = "partial"
    rejected = "rejected"


class TransferOrderStatus(str, enum.Enum):
    draft = "draft"
    approved = "approved"
    shipped = "shipped"
    received = "received"
    cancelled = "cancelled"


class EmploymentStatus(str, enum.Enum):
    trial = "trial"
    probation = "probation"
    regular = "regular"
    resigned = "resigned"


class EmploymentType(str, enum.Enum):
    regular = "regular"
    part_time = "part_time"
    intern = "intern"
    trainee = "trainee"
    temp = "temp"


class StorageType(str, enum.Enum):
    frozen = "frozen"
    chilled = "chilled"
    ambient = "ambient"
    live = "live"


class RFMLevel(str, enum.Enum):
    S1 = "S1"
    S2 = "S2"
    S3 = "S3"
    S4 = "S4"
    S5 = "S5"

File: shared/api-types/src/enums.ts
/**
 * 枚举类型 — 与 shared/ontology/src/enums.py 一一对应
 */

export type OrderStatus =
  | 'pending'
  | 'confirmed'
  | 'preparing'
  | 'ready'
  | 'served'
  | 'completed'
  | 'cancelled';

export type StoreStatus =
  | 'active'
  | 'inactive'
  | 'renovating'
  | 'preparing'
  | 'closed';

export type InventoryStatus =
  | 'normal'
  | 'low'
  | 'critical'
  | 'out_of_stock';

export type TransactionType =
  | 'purchase'
  | 'usage'
  | 'waste'
  | 'adjustment'
  | 'transfer'
  | 'receiving'
  | 'transfer_out'
  | 'transfer_in';

export type ReceivingOrderStatus =
  | 'draft'
  | 'inspecting'
  | 'partially_received'
  | 'fully_received'
  | 'rejected';

export type ReceivingItemStatus =
  | 'pending'
  | 'accepted'
  | 'partial'
  | 'rejected';

export type TransferOrderStatus =
  | 'draft'
  | 'approved'
  | 'shipped'
  | 'received'
  | 'cancelled';

export type EmploymentStatus =
  | 'trial'
  | 'probation'
  | 'regular'
  | 'resigned';

export type EmploymentType =
  | 'regular'
  | 'part_time'
  | 'intern'
  | 'trainee'
  | 'temp';

export type StorageType =
  | 'frozen'
  | 'chilled'
  | 'ambient'
  | 'live';

export type RFMLevel =
  | 'S1'
  | 'S2'
  | 'S3'
  | 'S4'
  | 'S5';

/** 订单类型（对应 Order.order_type） */
export type OrderType =
  | 'dine_in'
  | 'takeaway'
  | 'delivery'
  | 'retail'
  | 'catering'
  | 'banquet';

/** 门店类型（对应 Store.store_type） */
export type StoreType =
  | 'physical'
  | 'virtual'
  | 'central_kitchen'
  | 'warehouse';

/** 员工角色 */
export type EmployeeRole =
  | 'waiter'
  | 'chef'
  | 'cashier'
  | 'manager';

/** 定价模式（对应 OrderItem.pricing_mode） */
export type PricingMode =
  | 'fixed'
  | 'weight'
  | 'market_price';
